bin/NativeTests/JsRTApiTest.cpp

```diff
@@ -660,6 +660,215 @@ namespace JsRTApiTest
+    }
+
+    struct ExternaEnhancedBaseClassFunctionTestInfo
+    {
+        JsValueRef derived;
+        JsValueRef base;
+    };
+
+    JsValueRef CALLBACK ExternaEnhancedBaseClassFunctionTestCallback(JsValueRef callee, JsValueRef *arguments, unsigned short argumentCount, JsNativeFunctionInfo *info, void *callbackData)
+    {
+        REQUIRE(callbackData != nullptr);
+
+        ExternaEnhancedBaseClassFunctionTestInfo* testinfo = (ExternaEnhancedBaseClassFunctionTestInfo*)callbackData;
+        JsValueType t;
+        REQUIRE(JsGetValueType(testinfo->derived, &t) == JsNoError);
+        REQUIRE(t == JsFunction);
+        REQUIRE(JsGetValueType(testinfo->base, &t) == JsNoError);
+        REQUIRE(t == JsFunction);
+        REQUIRE(argumentCount == 2);
+
+        JsPropertyIdRef propId;
+        bool success = false;
+        JsValueRef _true;
+        REQUIRE(JsGetTrueValue(&_true) == JsNoError);
+        JsValueRef _false;
+        REQUIRE(JsGetFalseValue(&_false) == JsNoError);
+
+        REQUIRE(info->thisArg == arguments[0]);
+        REQUIRE(JsStrictEquals(_true, arguments[1], &success) == JsNoError);
+        REQUIRE(success);
+
+        REQUIRE(info->isConstructCall);
+        REQUIRE(JsGetValueType(info->newTargetArg, &t) == JsNoError);
+        REQUIRE(t == JsFunction);
+        REQUIRE(JsGetValueType(info->thisArg, &t) == JsNoError);
+        REQUIRE(t == JsObject);
+
+        // new.target === Derived
+        REQUIRE(JsStrictEquals(info->newTargetArg, testinfo->derived, &success) == JsNoError);
+        REQUIRE(success);
+
+        // this.constructor === Derived
+        REQUIRE(JsGetPropertyIdFromName(_u("constructor"), &propId) == JsNoError);
+        JsValueRef thisCtor = JS_INVALID_REFERENCE;
+        REQUIRE(JsGetProperty(info->thisArg, propId, &thisCtor) == JsNoError);
+        REQUIRE(JsStrictEquals(thisCtor, testinfo->derived, &success) == JsNoError);
+        REQUIRE(success);
+
+        // this.__proto__ === Derived.prototype
+        JsValueRef thisProto = JS_INVALID_REFERENCE;
+        REQUIRE(JsGetPrototype(info->thisArg, &thisProto) == JsNoError);
+        JsValueRef derivedPrototype = JS_INVALID_REFERENCE;
+        REQUIRE(JsGetPropertyIdFromName(_u("prototype"), &propId) == JsNoError);
+        REQUIRE(JsGetProperty(testinfo->derived, propId, &derivedPrototype) == JsNoError);
+        REQUIRE(JsStrictEquals(thisProto, derivedPrototype, &success) == JsNoError);
+        REQUIRE(success);
+
+        return info->thisArg;
+    }
+
+    void ExternalEnhancedBaseClassFunctionTest(JsRuntimeAttributes attributes, JsRuntimeHandle runtime)
+    {
+        ExternaEnhancedBaseClassFunctionTestInfo info;
+
+        JsValueRef name;
+        REQUIRE(JsCreateString("BaseClass", 10, &name) == JsNoError);
+        JsValueRef base = JS_INVALID_REFERENCE;
+        REQUIRE(JsCreateEnhancedFunction(ExternaEnhancedBaseClassFunctionTestCallback, name, &info, &base) == JsNoError);
```

Comment:
Prefast complains that `info` is uninitialized here

Reply:
Thanks, have fixed this one.